pkg/store/bucket.go

```diff
@@ -651,6 +681,8 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 		}
 		blocks := bs.getFor(req.MinTime, req.MaxTime, req.MaxResolutionWindow)
 
+		debugFoundBlockSetOverview(s.logger, req.MinTime, req.MaxTime, bs.labels, blocks)
```

Comment:
Would it not be better putting some of the logging in the loop below otherwise we are iterating over the blocks twice?

Reply:
I think we will add some if to make sure it runs only on debug log level

Reply:
and it is not that costly. It runs only through the chosen blocks. (`getFor` runs through ALL blocks), so this one does not matter much. (no complexity change)